### ISSUE TITLE
bad results

### ISSUE THREAD

**Issue by ak9250** [shaoanlu/fewshot-face-translation-GAN#2]
i tried these two images but the result was like this
![badresult](https://user-images.githubusercontent.com/6818285/62001795-61a92900-b0c6-11e9-9d6e-5a459deb2f1b.png)
![Screen Shot 2019-07-27 at 11 28 24 PM](https://user-images.githubusercontent.com/6818285/62001800-6e2d8180-b0c6-11e9-8108-ee48297c52e6.png)

**Comment by shaoanlu** [comment]
The iris detector I used in this project does not perform well due to [its deviated pre-processing from official implementation](https://github.com/shaoanlu/GazeML-keras#wip). Disabling `draw_iris()` function in [`utils.py`](https://github.com/shaoanlu/fewshot-face-translation-GAN/blob/master/utils/utils.py) might help.

Also, the face alignment provided in this repo is a little bit different from my dev environment, where MTCNN, instead of S3FD+FAN, was used.

**Comment by ak9250** [comment]
@shaoanlu ok, can you please add mtcnn, i think it also reducing flickering as shown in your other face-swap repo. I will try it again by disabling draw_iris() thanks

**Comment by ak9250** [comment]
tried it again
![Screen Shot 2019-08-29 at 11 46 12 AM](https://user-images.githubusercontent.com/6818285/63955488-c6073180-ca52-11e9-8a54-24d7e9259966.png)
 I changed the return aligned_face, colored_parsing_map, aligned_im, (x0, y0, x1, y1), landmarks
to return the colored_parsing_map instead of the parsing_map_with_iris

**Comment by ak9250** [comment]
@shaoanlu also noticed occlusions cause jitter as well as different skin colors with source and target dont always result in a good swap
![outcombined (2)](https://user-images.githubusercontent.com/6818285/64228831-0e40ac80-ceb6-11e9-8d54-30af36092cbd.gif)

**Comment by gstark0** [comment]
@ak9250 How did you change the resolution and manage to swap faces on a GIF you posted?

**Comment by ak9250** [comment]
@gstark0 i used the google colab notebook here
see PR https://github.com/shaoanlu/fewshot-face-translation-GAN/pull/4#issue-313354486
and this is the link to the notebook
https://github.com/ak9250/fewshot-face-translation-GAN/blob/master/colab_demo.ipynb
I have been getting some good results but trying to see how to improve it
![ezgif com-resize (7)](https://user-images.githubusercontent.com/6818285/64275629-0b74a480-cf14-11e9-850a-1ae32d300e33.gif)

**Comment by gstark0** [comment]
@ak9250 Thanks! That's what I was actually looking for. Great job!

**Comment by gstark0** [comment]
BTW, any idea why #5 happens to me?

**Comment by gstark0** [comment]
@ak9250  BTW, any idea why #5 happens to me?

**Comment by ak9250** [comment]
@gstark0 is that happening in the google colab notebook? Are you able to get a result?

**Comment by gstark0** [comment]
@ak9250 I'm using Jupyter Notebook installed locally to run it but I guess it shouldn't make a difference. I'm not able to get a result, in fact I'm not even able to load images.

**Comment by ak9250** [comment]
@gstark0 go to
First, you'll need to enable GPUs for the notebook:
Navigate to Edit→Notebook Settings
select GPU from the Hardware Accelerator drop-down
I have not tested it in a local environment, I am using the gpu in google colab

**Comment by gstark0** [comment]
@ak9250 Unfortunately I already have the accelerator set to GPU, seems more like something with image dimensions but afaik it should run on any size, right?

**Comment by gstark0** [comment]
@ak9250 Could you upload sample source and single sample raw image you used for training? I'm wondering if this will make any difference.

**Comment by ak9250** [comment]
@gstark0 images I get were various sizes and from youtube videos and gifs. Can you share the images you are using to see if i can test what the problem is?

**Comment by gstark0** [comment]
@ak9250 I've used these files --> https://filebin.net/l36dca5kb00hh93e <-- They're pretty much first results from Google just to test things and see if it will actually run, I don't care about accuracy for now. Nico as a source and Trump as a target.

**Comment by ak9250** [comment]
@gstark0 although the result isnt good, I was able to do a test with nicolas cage and donald trump, maybe try another set of images?
![outputstack (2)](https://user-images.githubusercontent.com/6818285/64284307-5350f700-cf27-11e9-804e-21eef2da5489.gif)

**Comment by gstark0** [comment]
Already tried, still the same ;/

**Comment by ak9250** [comment]
@gstark0 are you using youtube-dl to get the video and split it into frames with ffmpeg or some other method?
where is your test image coming from?

**Comment by gstark0** [comment]
@ak9250 No I don’t, for now. I just wanted to run it first and make sure everything works correctly. The images come from the internet, found in Google. Funny thing is - I checked the exact same images on Google colaba and it works fine. I guess it may be because of different TensorFlow version or some other library. Will check this tomorrow.

**Comment by shaoanlu** [comment]
The unnatural contrast and highlight in the output faces are the intrinsic characteristics of the released model. These artifacts can also be observed in the readme figures. I did not spend much time taking care of the training process and tuning objective functions but anyway this is the status quo of how the current model performs.

**Comment by ak9250** [comment]
@shaoanlu anyway the model can be improved? I am looking into training the model if the training code is released and release the trained model. Also is the training process similar to FUNIT as they do pet swapping instead of faces with good results but it is trained for 2 weeks on 8 v100 gpus with a dataset of about 100k animals

**Comment by gstark0** [comment]
@shaoanlu So when can we expect the training code to be released? We’d like to experiment more and improve the model :)

**Comment by shaoanlu** [comment]
I might not have to time to touch the code until mid or late Oct. 😔.

**Comment by ak9250** [comment]
@shaoanlu ok, any way I can help?

**Comment by zhulei1228** [comment]
> @shaoanlu anyway the model can be improved? I am looking into training the model if the training code is released and release the trained model. Also is the training process similar to FUNIT as they do pet swapping instead of faces with good results but it is trained for 2 weeks on 8 v100 gpus with a dataset of about 100k animals

Have you got better results?

**Comment by zhulei1228** [comment]
> > > @shaoanlu anyway the model can be improved? I am looking into training the model if the training code is released and release the trained model. Also is the training process similar to FUNIT as they do pet swapping instead of faces with good results but it is trained for 2 weeks on 8 v100 gpus with a dataset of about 100k animals
> > 
> > 
> > Have you got better results?
> 
> I got some decent results
> ![outputstack2 (2)](https://user-images.githubusercontent.com/6818285/64836703-281a7780-d5b9-11e9-9a5d-f93099e7d8c3.gif)
It looks good except for the eyes.What changes do you made in this code?

**Comment by shaoanlu** [comment]
The author of FUNIT actually showed results on face identity translation in [this video](https://youtu.be/IhAsXcCz8LI?t=1220). 

I've also trained a FUNIT model on VGGFace2 dataset using the official implementation. The figure below is the results after ~90k iters of training. The 1st and 4th rows are real images.

![gen_train_00087500_03](https://user-images.githubusercontent.com/22006284/65003524-d1ec6200-d933-11e9-86ae-8520958c5e5c.jpg)

**Comment by ak9250** [comment]
> > > > @shaoanlu anyway the model can be improved? I am looking into training the model if the training code is released and release the trained model. Also is the training process similar to FUNIT as they do pet swapping instead of faces with good results but it is trained for 2 weeks on 8 v100 gpus with a dataset of about 100k animals
> > > 
> > > 
> > > Have you got better results?
> > 
> > 
> > I got some decent results
> > ![outputstack2 (2)](https://user-images.githubusercontent.com/6818285/64836703-281a7780-d5b9-11e9-9a5d-f93099e7d8c3.gif)
> > It looks good except for the eyes.What changes do you made in this code?

I have not made any changes other than https://github.com/shaoanlu/fewshot-face-translation-GAN/pull/4#issue-313354486

**Comment by ak9250** [comment]
@shaoanlu great, are you planning to release the training code for this repo or update the pretrained models?

**Comment by gstark0** [comment]
@shaoanlu When will the code be released?
